propagation.md

```diff
@@ -0,0 +1,145 @@
+
+This document provides a concise summary of the design and proper use of `Inject` and `Join`, regardless of the particular OpenTracing language or OpenTracing implementation.
+
+## "The Big Picture" for explicit trace propagation
+
+The hardest thing about distributed tracing is the *distributed* part. Any tracing system needs a way of understanding the causal relationship between activities in many distinct processes, whether they be connected via formal RPC frameworks, pub-sub systems, generic message queues, ad hoc HTTP calls, best-effort UDP packets, or something else entirely.
+
+Some distributed tracing systems (e.g., [Project5](http://dl.acm.org/citation.cfm?id=945454) from 2003, or [WAP5](http://www2006.org/programme/item.php?id=2033) from 2006 or [The Mystery Machine](https://www.usenix.org/node/186168) from 2014) *infer* causal relationships across process boundaries. Of course **there is a tradeoff between the apparent convenience of these black-box inference-based approaches and the ultimate freshness and quality of the assembled traces.** Per the concern about quality, OpenTracing is an *explicit* distributed tracing instrumentation standard, and as such it is much better-aligned with approaches like [X-Trace](https://www.usenix.org/conference/nsdi-07/x-trace-pervasive-network-tracing-framework) from 2007, [Dapper](http://research.google.com/pubs/pub36356.html) from 2010, or numerous open-source tracing systems like [Zipkin](https://github.com/openzipkin) or [Appdash](https://github.com/sourcegraph/appdash) (among others).
+
+Let's briefly restate the "Why OpenTracing?" section of the [OpenTracing landing page](/):
+
+> By offering consistent, expressive, vendor-neutral APIs for popular platforms, OpenTracing makes it easy for developers to add (or switch) tracing implementations with an O(1) configuration change. OpenTracing also offers a lingua franca for OSS instrumentation and platform-specific tracing helper libraries.
+
+**Together, `Inject` and `Join` allow for inter-process trace propagation without tightly coupling the programmer to a particular OpenTracing implementation.** 
+
+## Requirements for the OpenTracing propagation scheme
+
+For `Inject` and `Join` to be useful, all of the following must be true:
+
+- Per the above, the [OpenTracing user](/use-cases/#stepping-back-who-is-opentracing-for) handling cross-process trace propagation must not need to write OpenTracing-implementation-specific code
+- OpenTracing implementations must not need special handlers for every known inter-process communication mechanism: that's far too much work, and it's not even well-defined
+- That said, the propagation mechanism should be extensible for optimizations
+
+## The basic approach
+
+Any Span in a trace may be **Injected** into a **Carrier**. Similarly, given a **Carrier**, an injected trace may be **Joined**, yielding a new Span.
```

Comment:
I think we should first define the "carrier" being an abstraction of the IPC data exchange mechanism